core/js/sharedialogshareelistview.js

```diff
@@ -16,90 +16,84 @@
 		'<ul id="shareWithList" class="shareWithList">' +
 		'	{{#each sharees}}' +
-		'	<li class="shareWithList__item" data-share-id="{{shareId}}" data-share-type="{{shareType}}" data-share-with="{{shareWith}}">' +
-		'		<div class="shareWithList__container--left">' +
-		'			{{#if avatarEnabled}}' +
-		'			<div class="avatar {{#if modSeed}}imageplaceholderseed{{/if}}" data-username="{{shareWith}}" {{#if modSeed}}data-seed="{{shareWith}} {{shareType}}"{{/if}}></div>' +
+		'	<li data-share-id="{{shareId}}" data-share-type="{{shareType}}" data-share-with="{{shareWith}}">' +
+		'		<a href="#" class="unshare"><span class="icon-loading-small hidden"></span><span class="icon icon-delete"></span><span class="hidden-visually">{{unshareLabel}}</span></a>' +
+		'		<a href="#" class="toggleShareDetails"><span class="icon icon-settings-dark"></span><span class="hidden-visually">{{unshareLabel}}</span></a>' +
+		'		{{#if expirationDate}}' +
+		'		<a class="time"><span class="icon icon-time"></span><span class="hidden-visually">{{unshareLabel}}</span></a>' +
+		'		{{/if}}' +
+		'		{{#if avatarEnabled}}' +
+		'		<div class="avatar {{#if modSeed}}imageplaceholderseed{{/if}}" data-username="{{shareWith}}" {{#if modSeed}}data-seed="{{shareWith}} {{shareType}}"{{/if}}></div>' +
+		'		{{/if}}' +
+		'		<span class="has-tooltip username" title="{{shareWith}}">{{shareWithDisplayName}}</span>' +
+		'		{{#if shareWithAdditionalInfo}}' +
+		'		<span class="has-tooltip user-additional-info">({{shareWithAdditionalInfo}})</span>' +
+		'		{{/if}}' +
+		'		{{#if mailNotificationEnabled}}  {{#unless isRemoteShare}}' +
+		'		<span class="shareOption">' +
+		'		{{#unless wasMailSent}}' +
+		'		<span class="mailNotificationSpinner icon-loading-small hidden"></span>' +
+		'		<input id="mail-{{cid}}-{{shareWith}}" type="button" name="mailNotification" value="{{notifyByMailLabel}}" class="mailNotification checkbox" />' +
+		'		{{/unless}}' +
+		'		</span>' +
+		'		{{/unless}} {{/if}}' +
+		'		<div class="expirationOption">' +
+		'			{{#if isUserShare}}' +
+		'			<label for="expiration-{{name}}-{{cid}}-{{shareWith}}">Expiration: ' +
+		'				<input type="text" id="expiration-{{name}}-{{cid}}-{{shareWith}}" value="{{expirationDate}}" class="expiration expiration-user" placeholder="Choose an expiration date" />' +
+		'				<button class="removeExpiration">Remove</button>' +
+		'			</label>' +
 		'			{{/if}}' +
-		'			<span class="has-tooltip username" title="{{shareWith}}">{{shareWithDisplayName}}</span>' +
-		'		</div>' + 
-		'		<div class="shareWithList__container--right">'+
-		'			{{#if expirationDate}}' +
-		'			<div class="action-item"><span class="icon icon-time"></span><span class="hidden-visually">{{unshareLabel}}</span></div>' +
+		'			{{#if isGroupShare}}' +
+		'			<label for="expiration-{{name}}-{{cid}}-{{shareWith}}">Expire share on: ' +
+		'				<input type="text" id="expiration-{{name}}-{{cid}}-{{shareWith}}" value="{{expirationDate}}" class="expiration expiration-group"/>' +
```

Comment:
```suggestion
		'				<input type="text" id="expiration-{{name}}-{{cid}}-{{shareWith}}" value="{{expirationDate}}" class="expiration expiration-group" placeholder="Choose an expiration date" />' +
```
also IMO this should be the same as for a user

Reply:
fixed